File: frontend/src/types/overview.ts
export interface OverviewSeriesPoint {
  key: string;
  label: string;
  value: number;
}

export interface OperationalRecentTask {
  id: string;
  project_id: string;
  project_name: string;
  title: string;
  status: string;
  priority: string;
  assignee_id?: string | null;
  assignee_name?: string | null;
  assignee_avatar?: string | null;
  due_date?: string | null;
  updated_at: string;
}

export interface OperationalOverview {
  total_projects: number;
  active_tasks: number;
  overdue_tasks: number;
  team_members: number;
  completed_by_week: OverviewSeriesPoint[];
  recent_tasks: OperationalRecentTask[];
}

export interface FinanceOverviewPoint {
  key: string;
  label: string;
  income: number;
  outcome: number;
}

export interface HrisUpcomingRenewal {
  id: string;
  name: string;
  vendor: string;
  renewal_date: string;
  days_remaining: number;
  cost_amount: number;
  cost_currency: string;
  pic_employee_name?: string | null;
}

export interface HrisRecentReimbursement {
  id: string;
  employee_id: string;
  employee_name: string;
  title: string;
  category: string;
  amount: number;
  transaction_date: string;
  description: string;
  status: string;
  attachments: string[];
  created_at: string;
  updated_at: string;
}

export interface HrisOverview {
  total_employees: number;
  active_subscriptions: number;
  active_subscription_monthly_cost: number;
  monthly_net: number;
  pending_reimbursements: number;
  income_vs_outcome: FinanceOverviewPoint[];
  upcoming_renewals: HrisUpcomingRenewal[];
  recent_reimbursements: HrisRecentReimbursement[];
}

export interface MarketingRoasTrendPoint {
  key: string;
  label: string;
  spent: number;
  revenue: number;
  roas?: number | null;
}

export interface MarketingLeadStage {
  status: string;
  label: string;
  lead_count: number;
  estimated_value: number;
}

export interface MarketingTopCampaign {
  campaign_id: string;
  campaign_name: string;
  status: string;
  total_spent: number;
  total_revenue: number;
  roas?: number | null;
}

export interface MarketingOverview {
  active_campaigns: number;
  total_ads_spent: number;
  overall_roas?: number | null;
  total_leads: number;
  conversion_rate: number;
  roas_trend: MarketingRoasTrendPoint[];
  leads_by_stage: MarketingLeadStage[];
  top_campaigns: MarketingTopCampaign[];
}

File: backend/internal/model/overview.go
package model

import "time"

type OverviewSeriesPoint struct {
	Key   string `json:"key"`
	Label string `json:"label"`
	Value int64  `json:"value"`
}

type OperationalRecentTask struct {
	ID             string     `json:"id"`
	ProjectID      string     `json:"project_id"`
	ProjectName    string     `json:"project_name"`
	Title          string     `json:"title"`
	Status         string     `json:"status"`
	Priority       string     `json:"priority"`
	AssigneeID     *string    `json:"assignee_id,omitempty"`
	AssigneeName   *string    `json:"assignee_name,omitempty"`
	AssigneeAvatar *string    `json:"assignee_avatar,omitempty"`
	DueDate        *time.Time `json:"due_date,omitempty"`
	UpdatedAt      time.Time  `json:"updated_at"`
}

type OperationalOverview struct {
	TotalProjects   int64                   `json:"total_projects"`
	ActiveTasks     int64                   `json:"active_tasks"`
	OverdueTasks    int64                   `json:"overdue_tasks"`
	TeamMembers     int64                   `json:"team_members"`
	CompletedByWeek []OverviewSeriesPoint   `json:"completed_by_week"`
	RecentTasks     []OperationalRecentTask `json:"recent_tasks"`
}

type FinanceOverviewPoint struct {
	Key     string `json:"key"`
	Label   string `json:"label"`
	Income  int64  `json:"income"`
	Outcome int64  `json:"outcome"`
}

type HrisUpcomingRenewal struct {
	ID              string    `json:"id"`
	Name            string    `json:"name"`
	Vendor          string    `json:"vendor"`
	RenewalDate     time.Time `json:"renewal_date"`
	DaysRemaining   int       `json:"days_remaining"`
	CostAmount      int64     `json:"cost_amount"`
	CostCurrency    string    `json:"cost_currency"`
	PICEmployeeName *string   `json:"pic_employee_name,omitempty"`
}

type HrisOverviewSalaryHistoryRow struct {
	EmployeeID         string
	EffectiveDate      time.Time
	NetSalaryEncrypted string
}

type HrisOverviewSubscriptionRow struct {
	StartDate    time.Time
	BillingCycle string
	CostAmount   int64
}

type HrisOverview struct {
	TotalEmployees                int64                  `json:"total_employees"`
	TotalMonthlyPayroll           int64                  `json:"total_monthly_payroll"`
	ActiveSubscriptions           int64                  `json:"active_subscriptions"`
	ActiveSubscriptionMonthlyCost int64                  `json:"active_subscription_monthly_cost"`
	MonthlyNet                    int64                  `json:"monthly_net"`
	PendingReimbursements         int64                  `json:"pending_reimbursements"`
	MonthlyReimbursementTotal     int64                  `json:"monthly_reimbursement_total"`
	IncomeVsOutcome               []FinanceOverviewPoint `json:"income_vs_outcome"`
	UpcomingRenewals              []HrisUpcomingRenewal  `json:"upcoming_renewals"`
	RecentReimbursements          []Reimbursement        `json:"recent_reimbursements"`
}

type MarketingROASTrendPoint struct {
	Key     string   `json:"key"`
	Label   string   `json:"label"`
	Spent   int64    `json:"spent"`
	Revenue int64    `json:"revenue"`
	ROAS    *float64 `json:"roas,omitempty"`
}

type MarketingTopCampaign struct {
	CampaignID   string   `json:"campaign_id"`
	CampaignName string   `json:"campaign_name"`
	Status       string   `json:"status"`
	TotalSpent   int64    `json:"total_spent"`
	TotalRevenue int64    `json:"total_revenue"`
	ROAS         *float64 `json:"roas,omitempty"`
}

type MarketingOverview struct {
	ActiveCampaigns int64                     `json:"active_campaigns"`
	TotalAdsSpent   int64                     `json:"total_ads_spent"`
	OverallROAS     *float64                  `json:"overall_roas,omitempty"`
	TotalLeads      int64                     `json:"total_leads"`
	ConversionRate  float64                   `json:"conversion_rate"`
	ROASTrend       []MarketingROASTrendPoint `json:"roas_trend"`
	LeadsByStage    []LeadSummaryRow          `json:"leads_by_stage"`
	TopCampaigns    []MarketingTopCampaign    `json:"top_campaigns"`
}
